test/Confluent.SchemaRegistry.Serdes.IntegrationTests/Tests/ProduceConsume.cs

```diff
@@ -118,7 +115,63 @@ public static void ProduceConsume(string bootstrapServers, string schemaRegistry
 
                 consumer.Close();
             }
+
+            // Check that what's in schema registry is what's expected.
+            using (var schemaRegistry = new CachedSchemaRegistryClient(schemaRegistryConfig))
+            {
+                var subjects = schemaRegistry.GetAllSubjectsAsync().Result;
+
+                if (nameStrategy == SubjectNameStrategy.TopicRecord)
+                {
+                    Assert.Equal(2, (int)subjects.Where(s => s.Contains(topic)).Count());
+                    Assert.Single(subjects.Where(s => s == $"{topic}-key"));
+                    Assert.Single(subjects.Where(s => s == $"{topic}-{((Avro.RecordSchema)ProduceConsumeUser._SCHEMA).Fullname}"));
+                }
+
+                if (nameStrategy == SubjectNameStrategy.Topic)
+                {
+                    Assert.Equal(2, (int)subjects.Where(s => s.Contains(topic)).Count());
+                    Assert.Single(subjects.Where(s => s == $"{topic}-key"));
+                    Assert.Single(subjects.Where(s => s == $"{topic}-value"));
+                }
+
+                if (nameStrategy == SubjectNameStrategy.Record)
+                {
+                    Assert.Single(subjects.Where(s => s.Contains(topic))); // the string key.
+                    Assert.Single(subjects.Where(s => s == $"{topic}-key"));
+                    Assert.Single(subjects.Where(s => s == $"{((Avro.RecordSchema)ProduceConsumeUser._SCHEMA).Fullname}"));
+                }
+            }
         }
 
+        /// <summary>
+        ///     Test that messages produced with the Avro serializer can be consumed with the
+        ///     Avro deserializer (topic name strategy)
+        /// </summary>
+        [Theory, MemberData(nameof(TestParameters))]
+        private static void ProduceConsume_Topic(string bootstrapServers, string schemaRegistryServers)
```

Comment:
I'm sure it's possible in xunit to parameterize a single test rather than have these 3 variants, but i was too lazy to look it up, and xunit is super inconvenient for integration tests anyway, and at some point i want to get rid of it.

Reply:
I'm all for test duplication, it is much better than the alternative.

Reply:
Duplication is good. :)

Reply:
I mean there's almost certainly a short hand way of driving xunit to have exactly the same test coverage as this, but i'm too lazy to look in the manual.